src/executionTimingsReporter.ts

```diff
@@ -0,0 +1,139 @@
+import { HighResTimer, calculateElapsedTime } from "./utils";
+import { EventEmitter } from "events";
+
+export type TimingsReport = {[stepName: string]: [number] | number};
+
+export const stepStartedEventName = 'stepStarted';
+export const milestoneReachedEventName = 'milestoneReached';
+
+export interface StepStartedEventArguments {
+    stepName: string;
+}
+
+export interface MilestoneReachedEventArguments {
+    milestoneName: string;
+}
+
+export interface ObservableEvents<T> { // Normally T will be an EventEmitter
```

Comment:
Then you can say `T extends EventEmitter`

Reply:
Changed

Reply:
Sorry, that was an older version of the code. Now T is just an interface, so we cannot make it extend EventEmitter. I changed this for:
// T is an interface that declares the on methods (listeners) that we can subscribe to